### DISCUSSION TITLE
Multiple windows with different menu not working as expected

### DISCUSSION THREAD

**Discussion by shalapatil** [tauri-apps/tauri#11990] [General]
I am working on tauri app with next js. 

        tauri::Builder::default()
            .setup(|app| {
                let app_handle = app.handle();

                // Create the "main" window with its menu
                let main_window = WindowBuilder::new(
                    &app_handle,
                    "main".to_string(),
                    tauri::WindowUrl::App("/".into()),
                )
                .menu(create_main_menu()) // Attach the main menu
                .visible(false) // Hidden by default
                .build()?;
                println!("Main window created: {:?}", main_window);
        
                // Create the "onboarding" window with its menu
                let onboarding_window = WindowBuilder::new(
                    &app_handle,
                    "onboarding".to_string(),
                    tauri::WindowUrl::App("/onboarding".into()),
                )
                .menu(create_onboarding_menu()) // Attach the onboarding menu
                .build()?;
                println!("Onboarding window created: {:?}", onboarding_window);
        
        
                Ok(())
            })
            .run(tauri::generate_context!())
            .expect("error while running Tauri application");

Onboarding window is created at last. onboarding menu is only getting shown for main window as well. 

What could be the issue? 

thank you!

**Comment by FabianLars** [comment]
closing in favor of https://github.com/tauri-apps/tauri/issues/11991
